### PR TITLE
fix: add custom field in Purchase Receipt Line Item and search it in supplier part name in Supplier Items

### PR DESCRIPTION
**Pull request by hitenshKharva**
…supplier part name in Supplier Items

Ref : [Asana Task](https://app.asana.com/0/1199082161983365/1200238400032848)

### CODE REVIEW

File: bloomstack_core/bloomstack_core/custom/purchase_receipt_item.json [hunk 1]
```diff
@@ -317,10 +317,225 @@
...
+   "name": "Purchase Receipt Item-purchase_order-print_hide",
+   "owner": "Administrator",
+   "parent": null,
+   "parentfield": null,
+   "parenttype": null,
+   "property": "print_hide",
+   "property_type": "Check",
+   "value": "0"
+  }
+ ],
```
Comment (hrwX) [inline bloomstack_core/bloomstack_core/custom/purchase_receipt_item.json]: Remove this

File: bloomstack_core/bloomtrace/transfer.py [hunk 2]
```diff
@@ -5,11 +5,19 @@ def create_purchase_receipt(transfer):
...
 	for item in transfer.get("items", []):
-		item_exists = frappe.get_all("Item", filters={"metrc_item_name": item.pop("product_name")}, fields=["name", "item_name"])
-		if item_exists:
+		item_code = frappe.db.sql("""
+		SELECT
+			item_supp.parent
+		FROM
+			`tabItem Supplier` as item_supp,
+		WHERE item_supp.supplier_part_number = %(supplier)s
+			""", {"supplier": item.get("product_name")}, as_dict=1)
```
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py]: - Use ORM

**Review by hrwX — CHANGES_REQUESTED** [review]

File: bloomstack_core/bloomstack_core/custom/purchase_receipt_item.json [hunk 3]
```diff
@@ -317,10 +317,225 @@
...
+   "modified_by": "Administrator",
+   "name": "Purchase Receipt Item-purchase_order-print_hide",
+   "owner": "Administrator",
+   "parent": null,
+   "parentfield": null,
+   "parenttype": null,
+   "property": "print_hide",
+   "property_type": "Check",
+   "value": "0"
+  }
```
Comment (hrwX) [inline bloomstack_core/bloomstack_core/custom/purchase_receipt_item.json]: Do not add any new code

File: bloomstack_core/bloomtrace/transfer.py (hunk 2 above)
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py]: Use ORM

**Review by hrwX — CHANGES_REQUESTED** [review]

File: bloomstack_core/bloomtrace/transfer.py [hunk 4]
```diff
@@ -5,11 +5,13 @@ def create_purchase_receipt(transfer):
 	transfer = frappe.parse_json(transfer)
 
 	for item in transfer.get("items", []):
-		item_exists = frappe.get_all("Item", filters={"metrc_item_name": item.pop("product_name")}, fields=["name", "item_name"])
-		if item_exists:
+		item_code =  frappe.db.get_all(("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields = ["parent"])
+		if item_code:
+			item_exists = frappe.get_all("Item", filters={"item_code": item_code}, fields=["name", "item_name"])
```
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py]:
```suggestion
```

File: bloomstack_core/bloomtrace/transfer.py [hunk 5]
```diff
@@ -5,11 +5,13 @@ def create_purchase_receipt(transfer):
...
 			item.update({
-				"item_code": item_exists[0].name,
-				"item_name": item_exists[0].item_name
+				"item_code"	: item_exists[0].name,
+				"item_name"	: item_exists[0].item_name,
+				"metrc_product_name": item.get("product_name")
```
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py]:
```suggestion
				"item_code"	: item.parent,
				"metrc_product_name": item.get("product_name")
```

File: bloomstack_core/bloomtrace/transfer.py [hunk 6]
```diff
@@ -5,11 +5,13 @@ def create_purchase_receipt(transfer):
 	transfer = frappe.parse_json(transfer)
 
 	for item in transfer.get("items", []):
-		item_exists = frappe.get_all("Item", filters={"metrc_item_name": item.pop("product_name")}, fields=["name", "item_name"])
-		if item_exists:
+		item_code =  frappe.db.get_all(("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields = ["parent"])
```
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py]:
```suggestion
		item =  frappe.db.get_all(("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields = ["parent"])
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: bloomstack_core/bloomtrace/transfer.py [hunk 7]
```diff
@@ -5,12 +5,11 @@ def create_purchase_receipt(transfer):
...
-		item_exists = frappe.get_all("Item", filters={"metrc_item_name": item.pop("product_name")}, fields=["name", "item_name"])
-		if item_exists:
-			item.update({
-				"item_code": item_exists[0].name,
-				"item_name": item_exists[0].item_name
-			})
+		item =  frappe.db.get_all(("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields = ["parent"])
+		item.update({
+			"item_code"	: item.parent,
+			"metrc_product_name": item.get("product_name")
```
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py:11]:
```suggestion
			"item_code": supplier_item and supplier_item[0].parent,
			"metrc_product_name": item.get("product_name")
```

File: bloomstack_core/bloomtrace/transfer.py [hunk 8]
```diff
@@ -5,12 +5,11 @@ def create_purchase_receipt(transfer):
 	transfer = frappe.parse_json(transfer)
 
 	for item in transfer.get("items", []):
-		item_exists = frappe.get_all("Item", filters={"metrc_item_name": item.pop("product_name")}, fields=["name", "item_name"])
-		if item_exists:
-			item.update({
-				"item_code": item_exists[0].name,
-				"item_name": item_exists[0].item_name
-			})
+		item =  frappe.db.get_all(("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields = ["parent"])
```
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py:8]:
```suggestion
		supplier_item =  frappe.db.get_all(("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields = ["parent"])
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: bloomstack_core/bloomtrace/transfer.py (hunk 7 above)
Comment (hrwX) [inline bloomstack_core/bloomtrace/transfer.py]:
```suggestion
		supplier_item = frappe.db.get_all("Item Supplier", filters={"supplier_part_no": item.get("product_name")}, fields=["parent"])
		item.update({
			"item_code": supplier_item[0].parent if supplier_item else None,
			"metrc_product_name": item.get("product_name")
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**